sdk/openai/openai/samples/cookbook/bringYourOwnData/index.js

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+// Importing the @azure/search-documents library
+const { SearchIndexClient, SearchClient, AzureKeyCredential, odata } = require("@azure/search-documents");
+
+// Importing the @azure/openai library
+const { OpenAIClient } = require("@azure/openai");
+
+// Importing the index definition and sample data
+const hotelData = require('./hotels.json');
+const indexDefinition = require('./hotels_quickstart_index.json');
+
+// Load the .env file if it exists
+require("dotenv").config();
+
+// Getting search endpoint, search admin Key, Azure OpenAI endpoint, Azure API Key, and chat deployment id from .env file
+const azureSearchEndpoint = process.env.SEARCH_API_ENDPOINT || "";
+const azureSearchAdminKey = process.env.SEARCH_API_KEY || "";
+const azureApiKey = process.env.AZURE_API_KEY || "";
+const openAiEndpoint = process.env.AZURE_OPENAI_ENDPOINT || "";
+const chatDeploymentId = process.env.AZURE_OPENAI_DEPLOYMENT_ID || "";
+
+async function main() {
+    console.log(`Running Azure Cognitive Search Javascript quickstart...`);
+    if (!azureSearchEndpoint || !azureSearchAdminKey || !azureApiKey || !openAiEndpoint || !chatDeploymentId) {
+        console.log("Make sure to set valid values for azureSearchEndpoint, azureSearchAdminKey, azureApiKey, openAiEndpoint, and chatDeploymentId with proper authorization.");
+        return;
+    }
+
+    // Creating an index client to create the search index
+    const indexClient = new SearchIndexClient(azureSearchEndpoint, new AzureKeyCredential(azureSearchAdminKey));
+
+    // Getting the name of the index from the index definition
+    const indexName = indexDefinition["name"];
+
+    console.log('Checking if index exists...');
+    await deleteIndexIfExists(indexClient, indexName);
+
+    console.log('Creating index...');
+    let index = await indexClient.createIndex(indexDefinition);
+    console.log(`Index named ${index.name} has been created.`);
+
+    // Creating a search client to upload documents and issue queries
+    const searchClient = indexClient.getSearchClient(indexName);
+
+    console.log('Uploading documents...');
+    let indexDocumentsResult = await searchClient.mergeOrUploadDocuments(hotelData['value']);
+    console.log(`Index operations succeeded: ${JSON.stringify(indexDocumentsResult.results[0].succeeded)} `);
+
+    // waiting one second for indexing to complete (for demo purposes only)
+    sleep(1000);
```

Comment:
Let's have this async so we're not even suggesting a bad practice of busy waiting the Node.js runtime.

Reply:
Updated with [`90f9611`](https://github.com/Azure/azure-sdk-for-js/pull/27298/commits/90f9611a8f53cd98e2ca46f12c1bfbd931261293)